s/String.cpp

```diff
@@ -49,11 +49,8 @@ extern "C" String* sStringReadLine(String *string) {
     return string;
 }
 
-extern "C" char sStringBeginsWith(String *string, String *beginning) {
-    if (string->count < beginning->count) {
-        return false;
-    }
-    return std::memcmp(string->characters.get(), beginning->characters.get(), beginning->count) == 0;
+extern "C" char sStringBeginsWithAtIndex(String *string, String *beginning, int utf8Index) {
```

Comment:
I think it should be runtime::Integer instead of int here. https://www.emojicode.org/docs/guides/api.html#function-signatures

Reply:
Also, I think it would be better to check `[string->characters.get() + utf8Index, string->characters.get() + utf8Index + beginning->count)` is in bounds here, as this method is exposed publicly below.